### PR TITLE
Issue #13100: Create tests from header category checks xdoc examples

### PR DESCRIPTION
**Pull request by stoyanK7**
Part of #13100 

https://checkstyle.org/checks/header/index.html

### CODE REVIEW

**Comment by stoyanK7** [comment]
Github, generate site

**Review by romani — APPROVED** [review]

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/header/regexpheader/Example3.txt [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+      name="header"
+      value="^// Copyright \(C\) (\d\d\d\d -)? 2004 MyCompany$
+        \n^// All rights reserved$"/>
+  </module>
+</module>
+*/
+
+// xdoc section -- start
+^// Copyright \(C\) (\d\d\d\d -)? 2004 MyCompany$
+^// All rights reserved$
```
Comment (nrmancuso) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/header/regexpheader/Example3.txt:14]: It is not obvious to me how this is translated to https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/620eec4_2023152225/checks/header/header.html#Example3-code.

**Review by nrmancuso — CHANGES_REQUESTED** [review]
Item:

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/header/regexpheader/Example3.txt (hunk 1 above)
Comment (romani) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/header/regexpheader/Example3.txt:14]: It is converted to https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/620eec4_2023152225/checks/header/regexpheader.html#Example3-code

You looked at different Check

Comment (nrmancuso) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/header/regexpheader/Example3.txt:14]: Ah, I see, thanks for pointing this out :)

**Review by nrmancuso — APPROVED** [review]

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/header/regexpheader/Example3.txt (hunk 1 above)
Comment (romani) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/header/regexpheader/Example3.txt:14]: https://github.com/checkstyle/checkstyle/issues/13585

**Pull request merged**